nomenklatura/resolver/resolver.py

```diff
@@ -1,90 +1,320 @@
-import logging
+#
+# Don't forget to call self._invalidate from methods that modify edges.
+#
 import getpass
-from pathlib import Path
-from threading import RLock
+import logging
 from functools import lru_cache
-from collections import defaultdict
-from typing import Dict, Generator, List, Optional, Set, Tuple
+from typing import Dict, Generator, Optional, Set, Tuple
+
+from followthemoney.types import registry
 from rigour.ids.wikidata import is_qid
 from rigour.time import utc_now
+from sqlalchemy import Column, Float, MetaData, Table, Unicode, alias, func, or_
+from sqlalchemy.engine import Connection, Engine, Transaction
+from sqlalchemy.sql.expression import select, update
 
+from nomenklatura.db import get_engine, get_upsert_func
 from nomenklatura.entity import CE
 from nomenklatura.judgement import Judgement
-from nomenklatura.resolver.identifier import Identifier, StrIdent, Pair
 from nomenklatura.resolver.edge import Edge
+from nomenklatura.resolver.identifier import Identifier, StrIdent
 from nomenklatura.resolver.linker import Linker
+from nomenklatura.statement.statement import Statement
 from nomenklatura.util import PathLike
 
 log = logging.getLogger(__name__)
 
 
+def timestamp() -> str:
+    return utc_now().isoformat()[:28]
+
+
 class Resolver(Linker[CE]):
     UNDECIDED = (Judgement.NO_JUDGEMENT, Judgement.UNSURE)
 
-    def __init__(self, path: Optional[Path] = None) -> None:
-        self.path = path
-        self.lock = RLock()
-        self.edges: Dict[Pair, Edge] = {}
-        self.nodes: Dict[Identifier, Set[Edge]] = defaultdict(set)
+    def __init__(
+        self,
+        engine: Engine,
+        metadata: MetaData,
+        create: bool = False,
+        table_name: str = "resolver",
```

Comment:
@pudo  configurable table name is unnecessary complication, right? 

Initially I thought we might want to be able to keep a table around but quickly switch to a new table to debug stuff. But we can just as easily rename that table, and load the data into a new `resolver`